bee-protocol/src/workers/metrics.rs

```diff
@@ -1,29 +1,74 @@
 // Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::{convert::Infallible, time::Duration};
+use std::{convert::Infallible, net::SocketAddr, time::Duration};
 
 use async_trait::async_trait;
 use bee_ledger::workers::event::{MilestoneConfirmed, PrunedIndex, SnapshottedIndex};
+use bee_metrics::{metrics::MemoryUsage, Registry};
 use bee_runtime::{node::Node, shutdown_stream::ShutdownStream, worker::Worker};
 use futures::StreamExt;
 use log::info;
+use serde::Deserialize;
 use tokio::time::interval;
 use tokio_stream::wrappers::IntervalStream;
 
 use crate::types::metrics::NodeMetrics;
 
 const METRICS_INTERVAL: Duration = Duration::from_secs(60);
 
+const DEFAULT_BIND_ADDRESS: &str = "0.0.0.0:3030";
+
+#[derive(Default, Deserialize, PartialEq)]
+#[must_use]
+pub struct MetricsWorkerConfigBuilder {
```

Comment:
Can we rename this to `MetricsConfigBuilder`? I feel like the term `worker` doesn't add much tbh, and is probably not future-proof since we may decide to drop the worker system for something else at some point.